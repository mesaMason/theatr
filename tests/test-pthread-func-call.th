func greeter() -> void:
    print("Hello from inside greeter thread")

func goodbye() -> void:
    print("Hello from inside goodbye thread")

func calculator(int c) -> void:
    print("Hello from inside calculator thread. My count is ")
    print(c)

func main() -> int:
    pthread_create(calculator, 5)
    pthread_create(greeter)
    pthread_create(goodbye)

    return 0
