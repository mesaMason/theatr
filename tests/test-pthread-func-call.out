Hello from inside calculator thread. My count is
5
Hello from inside greeter thread
Hello from inside goodbye thread
